asciidoctorj-core/src/main/java/org/asciidoctor/extension/Processor.java

```diff
@@ -16,25 +17,29 @@
 
 public class Processor {
 
-	protected RubyHash config;
-    protected Ruby rubyRuntime;
+	protected Map<String, Object> config;
 
     public Processor(Map<String, Object> config) {
-        this.rubyRuntime = JRubyRuntimeContext.get();
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
+        this.config = config;
     }
 
-    public void update_config(Map<Object, Object> config) {
+    public void update_config(Map<String, Object> config) {
     	this.config.putAll(config);
     }
     
-    public Map<Object, Object> getConfig() {
+    public Map<String, Object> getConfig() {
     	return this.config;
     }
 
-    public RubyHash setConfig(Map<String, Object> config) {
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
-        return null;
+    public void setConfig(Map<String, Object> config) {
+        if (config instanceof RubyHashMapDecorator) {
+            // ProcessorProxy has already taken over, replace config
+            this.config.clear();
+            this.config.putAll(config);
+        } else {
+            // Processor is still independent, simply set the member
+            this.config = config;
+        }
```

Comment:
At the moment I don't see how I can get around without setConfig().
To sync the Java part with the Ruby part the ProcessorProxy has to re-set the config after the Ruby part has been initialized.
(Remember that the Ruby part is always initialized after the Java part)

Maybe someone has an idea to make the config only settable from the ProcessorProxies?
They are in a different package so making them package private will not work either...

Any ideas are welcome.

It would also work out to simply overwrite the config member:
```java
public void setConfig(Map<String, Object> config) {
    this.config = config;
}
```
and assume that the API user does not invoke this method again after the constructor has finished.
Because during the constructor execution calling setConfig() is IMO perfectly valid.